### PR TITLE
`azurerm_data_factory_linked_service_azure_blob_storage` - Add support for `connection_string_insecure` 

### LINKED ISSUE #19862: Azurerm_data_factory_linked_service_azure_blob_storage account name stored asterisks
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.0

### AzureRM Provider Version

3.0

### Affected Resource(s)/Data Source(s)

azurerm_data_factory_linked_service_azure_blob_storage

### Terraform Configuration Files

```hcl
provider "azurerm" {
  # Configuration options
  features {}
}

data "azurerm_resource_group" "test" {
  name     = "test-lg"
 # location = "East US"
}

data "azurerm_client_config" "current" {
}

data "azurerm_storage_account" "example" {
  name                = "levgoronshteynstorage"
  resource_group_name = data.azurerm_resource_group.test.name
}

resource "azurerm_data_factory" "test" {
  name                = "example-lev"
  location            = data.azurerm_resource_group.test.location
  resource_group_name = data.azurerm_resource_group.test.name
}

resource "azurerm_key_vault" "test" {
  name                = "example-lev"
  location            = data.azurerm_resource_group.test.location
  resource_group_name = data.azurerm_resource_group.test.name
  tenant_id           = data.azurerm_client_config.current.tenant_id
  sku_name            = "standard"
}

resource "azurerm_data_factory_linked_service_key_vault" "test" {
  name            = "linkkv-lev"
  data_factory_id = azurerm_data_factory.test.id
  key_vault_id    = azurerm_key_vault.test.id
}

resource "azurerm_data_factory_linked_service_azure_blob_storage" "test" {
  name            = "example-lev"
  data_factory_id = azurerm_data_factory.test.id
 service_principal_id = "f38c0ec8-c8d0-4df8-b29b-69057f6236984239"
 service_principal_key = "6zc5e2d7-5f12-5669-9ea5-8108093a60ce"
 tenant_id = "0d4q3e88-8caf-41ca-b4da-e3b33b6c52ec"
 connection_string = "DefaultEndpointsProtocol=https;AccountName=levgoronshteynstorage;AccountKey=NJaByjNTH3HZcRyROBeDFPjKoPIUMViI0p7r+79tRLBVSQNmmspGBdwI4ns2qGyp1HtvrZC+6hYf+AStVaZ8eg==;EndpointSuffix=core.windows.net"

ALL sensitive information has been changed!
```

### Debug Output/Panic Output

```shell
No panic output but what occurs is when you login to Azure portal and navigate to Azure Data Factory to see what the blob storage account name is, you are not able to see anything but asterisks(********). If you perform the same steps manually via Azure Portal, you can see the blob storage account name.

Same issue accounts even if you use AZ CLI
```

### Expected Behaviour

When a user uses Azure Portal to add a linked service for blob storage, the portal breaks up the connection string into two properties below:
```
{
  "connectionString": "<value>",
  "encryptedCredential": "<value>"
}
```

### Actual Behaviour

When using AzureRM provider, the connection string is stored as secure string and the storage account name is hidden with asterisks.
```
"connectionString": {
      "type": "SecureString",
      "value": "**********"
    },
    "credential": null,
    "description": null,
    "encryptedCredential": "ew0KICAiVmVyc2lvbiI6ICIyMDE3LTExLTMwIiwNCiAgIlByb3RlY3Rpb25Nb2RlIjogIktleSIsDQogICJTZWNyZXRDb250ZW50VHlwZSI6ICJQbGFpbnRleHQiLA0KICAiQ3JlZGVudGlhbElkIjogIkRBVEFGQUNUT1JZQEREMjREMjZGLTFEQkYtNDVERi04NEM1LTIzRUI4MDIzMUU1QV9hZGU1YTkzYS00MWZlLTRjZGYtOTA1NS1lNzg5YzcwMzM2NDciDQp9",
```
### Steps to Reproduce

Use the Terraform Configuration File provided

change the resource group and storage account

apply the terraform file.

### Important Factoids

_No response_

### References

The AZ CLI tool has the same behaviour as terraform:

```
az datafactory linked-service create --factory-name "example-lev" --properties "{\"type\":\"AzureBlobStorage\",\"typeProperties\":{\"connectionString\":

{\"type\":\"SecureString\",\"value\":\"DefaultEndpointsProtocol=https;AccountName=xxx;AccountKey=xxx+xxx+6hYf+AStVaZ8eg==;EndpointSuffix=core.windows.net\"}}}" --name "exampleLinkedService" --resource-group "test-lg"
{
  "etag": "4f008de4-0000-0100-0000-63b477df0000",
  "id": "/subscriptions/xxx/resourceGroups/test-lg/providers/Microsoft.DataFactory/factories/example-lev/linkedservices/exampleLinkedService",
  "name": "exampleLinkedService",
  "properties": {
    "accountKey": null,
    "accountKind": null,
    "additionalProperties": null,
    "annotations": null,
    "azureCloudType": null,
    "connectVia": null,
    "connectionString": {
      "type": "SecureString",
      "value": "**********"
    },
    "credential": null,
    "description": null,
    "encryptedCredential": "xxxxxxxxxxxxxxxxxxxxxxxx",
    "parameters": null,
    "sasToken": null,
    "sasUri": null,
    "serviceEndpoint": null,
    "servicePrincipalId": null,
    "servicePrincipalKey": null,
    "tenant": null,
    "type": "AzureBlobStorage"
  },
  "resourceGroup": "test-lg",
  "type": "Microsoft.DataFactory/factories/linkedservices"

```

### PR DESCRIPTION
**Pull request by mbfrahry**
Fixes #19862

### CODE REVIEW

File: website/docs/r/data_factory_linked_service_azure_blob_storage.html.markdown [hunk 1]
```diff
@@ -114,9 +114,13 @@ The following supported arguments are common across all Azure Data Factory Linke
 
 The following supported arguments are specific to Azure Blob Storage Linked Service:
 
-* `connection_string` - (Optional) The connection string. Conflicts with `sas_uri` and `service_endpoint`.
+* `connection_string` - (Optional) The connection string. Conflicts with `insecure_connection_string`, `sas_uri` and `service_endpoint`.
 
-* `sas_uri` - (Optional) The SAS URI. Conflicts with `connection_string` and `service_endpoint`.
+* `insecure_connection_string` - (Optional) The connection string sent insecurely. Conflicts with `connection_string`, `sas_uri` and `service_endpoint`.
+
+~> **Note:** `connection_string` is sent to the Azure encrypted whilst `insecure_connection_string` is sent unencrypted. The benefit of using `insecure_connection_string` is that the storage account name is now viewable in state and portal.
```
Comment (katbyte) [inline website/docs/r/data_factory_linked_service_azure_blob_storage.html.markdown:121]: i think we should be more explicit here:
```suggestion
~> **Note:** `connection_string` uses the Azure [SecureString](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.management.datafactory.models.securestring) to encrypt the contents within the REST payload to azure whilst the  `insecure_connection_string` is sent as a regular string. Both properties are still sent using SSL/HTTPS. At this time the portal will not decrypt Secure Strings so the `connection_string` property will show as `******` whilst `insecure_connection_string` will be viewable in the portal.
```

File: internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource_test.go [hunk 2]
```diff
@@ -423,3 +438,28 @@ resource "azurerm_data_factory_linked_service_azure_blob_storage" "test" {
...
+resource "azurerm_data_factory" "test" {
+  name                = "acctestdf%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_data_factory_linked_service_azure_blob_storage" "test" {
+  name                        = "acctestlsblob%d"
+  data_factory_id            = azurerm_data_factory.test.id
+  insecure_connection_string = "DefaultEndpointsProtocol=https;AccountName=foo;AccountKey=bar"
```
Comment (katbyte) [inline internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource_test.go:462]: can we add a test that uses both?z	
```suggestion
  insecure_connection_string = "DefaultEndpointsProtocol=https;AccountName=foo;"
    connection_string = "AccountKey=bar"
```

File: internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource.go [hunk 3]
```diff
@@ -54,7 +55,20 @@ func resourceDataFactoryLinkedServiceAzureBlobStorage() *pluginsdk.Resource {
 				Optional:     true,
 				Sensitive:    true,
 				ValidateFunc: validation.StringIsNotEmpty,
-				ExactlyOneOf: []string{"connection_string", "sas_uri", "service_endpoint"},
+				ExactlyOneOf: []string{"connection_string", "insecure_connection_string", "sas_uri", "service_endpoint"},
+			},
+
+			"insecure_connection_string": {
```
Comment (katbyte) [inline internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource.go:61]: would this be better as
```suggestion
			"connection_string_insecure": {
```

**Review by katbyte — APPROVED** [review]
couple comments but LGTM

File: internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource_test.go (hunk 2 above)
Comment (mbfrahry) [reply internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource_test.go:462]: Unfortunately, we can't as we can only do one or the other. ConnectionString is an `interface{}` in the sdk which allows for you to send the connection string as a SecureString or a regular String

File: website/docs/r/data_factory_linked_service_azure_blob_storage.html.markdown (hunk 1 above)
Comment (mbfrahry) [reply website/docs/r/data_factory_linked_service_azure_blob_storage.html.markdown:121]: Agreed

**Pull request merged**